### PR TITLE
✨ Add condition types

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR implements the types defined in the [condition CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200506-conditions.md)

/area api

### CODE REVIEW

File: api/v1alpha3/condition_types.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	// ConditionSeverityWarning specifies that a condition with Status=False represents
+	// a warning about in process of transitioning to to the True state.
+	ConditionSeverityWarning ConditionSeverity = "Warning"
+
+	// ConditionSeverityInfo specifies that a condition with Status=False represents
+	// a information about the process of transitioning to to the True state.
+	ConditionSeverityInfo ConditionSeverity = "Info"
+
+	// ConditionSeverityNode should apply only to condition is in state "True".
+	ConditionSeverityNone ConditionSeverity = ""
```
Comment (vincepri) [inline api/v1alpha3/condition_types.go:43]:
```suggestion
	// ConditionSeverityNone should apply only to conditions with `Status=True`.
	ConditionSeverityNone ConditionSeverity = ""
```

File: api/v1alpha3/condition_types.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+)
+
+// ANCHOR: ConditionSeverity
+
+// ConditionSeverity expresses the severity of a Condition Type failing.
+type ConditionSeverity string
+
+const (
+	// ConditionSeverityError specifies that a condition with Status=False represents
+	// an error about in process of transitioning to to the True state.
```
Comment (vincepri) [inline api/v1alpha3/condition_types.go:31]:
```suggestion
	// ConditionSeverityError specifies that a condition with `Status=False` is an error.
```

File: api/v1alpha3/condition_types.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	ConditionSeverityError ConditionSeverity = "Error"
+
+	// ConditionSeverityWarning specifies that a condition with Status=False represents
+	// a warning about in process of transitioning to to the True state.
```
Comment (vincepri) [inline api/v1alpha3/condition_types.go:35]:
```suggestion
	// ConditionSeverityWarning specifies that a condition with `Status=False` is a warning.
```

File: api/v1alpha3/condition_types.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	ConditionSeverityWarning ConditionSeverity = "Warning"
+
+	// ConditionSeverityInfo specifies that a condition with Status=False represents
+	// a information about the process of transitioning to to the True state.
+	ConditionSeverityInfo ConditionSeverity = "Info"
```
Comment (vincepri) [inline api/v1alpha3/condition_types.go:40]:
```suggestion
	// ConditionSeverityInfo specifies that a condition with `Status=False` is informative.
	ConditionSeverityInfo ConditionSeverity = "Info"
```

File: api/v1alpha3/condition_types.go [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+// ANCHOR: ConditionType
+
+// ConditionType is a valid value for ClusterCondition.Type.
+type ConditionType string
+
+// Common ConditionTypes used by Cluster API objects.
+const (
+	// ReadyCondition defines the Ready condition type that summarizes the operational state of a Cluster API object.
+	ReadyCondition ConditionType = "Ready"
+)
```
Comment (vincepri) [inline api/v1alpha3/condition_types.go:57]: Would be great to move these to `condition_consts.go`

Comment (fabriziopandini) [reply api/v1alpha3/condition_types.go:57]: done!

**Comment by vincepri** [comment]
LGTM pending squash

It's happening 🎉 🎉 🎉 🎉

File: api/v1alpha3/condition_types.go [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+	// ConditionSeverityNone should apply only to conditions with `Status=True`.
+	ConditionSeverityNone ConditionSeverity = ""
+)
+
+// ANCHOR_END: ConditionSeverity
+
+// ANCHOR: ConditionType
+
+// ConditionType is a valid value for ClusterCondition.Type.
```
Comment (sedefsavas) [inline api/v1alpha3/condition_types.go:47]:
```suggestion
// ConditionType is a valid value for Condition.Type.
```

Comment (fabriziopandini) [reply api/v1alpha3/condition_types.go]: good catch!

**Comment by fabriziopandini** [comment]
squash!

**Pull request merged**